transports/janus_mqtt.c

```diff
@@ -427,6 +465,50 @@ int janus_mqtt_init(janus_transport_callbacks *callback, const char *config_path
 				JANUS_LOG(LOG_ERR, "Invalid publish-qos value: %s (falling back to default)\n", qos_item->value);
 				ctx->publish.qos = 1;
 			}
+
+		#ifdef MQTTVERSION_5
+			if (ctx->connect.mqtt_version == MQTTVERSION_5) {
+				/* MQTT 5 specific configuration */
+				janus_config_array *proxy_transaction_user_properties_array = janus_config_get(config, config_general, janus_config_type_array, "proxy_transaction_user_properties");
+				if(proxy_transaction_user_properties_array) {
+					GList *proxy_transaction_user_properties_array_items = janus_config_get_items(config, proxy_transaction_user_properties_array);
+					if(proxy_transaction_user_properties_array_items != NULL) {
+						int proxy_transaction_user_properties_array_len = g_list_length(proxy_transaction_user_properties_array_items);
+						if(proxy_transaction_user_properties_array_len > 0) {
+							ctx->publish.proxy_transaction_user_properties = g_array_sized_new(FALSE, FALSE, sizeof(char*), proxy_transaction_user_properties_array_len);
+
+							g_list_foreach(
+								proxy_transaction_user_properties_array_items,
+								(GFunc)janus_mqtt_set_proxy_transaction_user_property,
+								(gpointer)ctx->publish.proxy_transaction_user_properties
+							);
+						}
+					}
+				}
+
+				janus_config_array *add_transaction_user_properties_array = janus_config_get(config, config_general, janus_config_type_array, "add_transaction_user_properties");
+				if(add_transaction_user_properties_array) {
+					GList *add_transaction_user_properties_array_items = janus_config_get_arrays(config, add_transaction_user_properties_array);
+					if(add_transaction_user_properties_array_items != NULL) {
+						int add_transaction_user_properties_array_len = g_list_length(add_transaction_user_properties_array_items);
+						if(add_transaction_user_properties_array_len > 0) {
+							ctx->publish.add_transaction_user_properties = g_array_sized_new(FALSE, FALSE, sizeof(MQTTProperty), add_transaction_user_properties_array_len);
+
+							janus_mqtt_set_add_transaction_user_property_user_data user_data = {
+								ctx->publish.add_transaction_user_properties,
+								config
+							};
+
```

Comment:
Same here, the two lines above seem to isolate the above call for some unknown reason?